src/credentials/attestation_verifier/TestDACRevocationDelegateImpl.cpp

```diff
@@ -52,6 +85,56 @@ void TestDACRevocationDelegateImpl::ClearDeviceAttestationRevocationSetPath()
     mDeviceAttestationRevocationSetPath = mDeviceAttestationRevocationSetPath.substr(0, 0);
 }
 
+CHIP_ERROR TestDACRevocationDelegateImpl::GetSubjectAndKeyIdFromPEMCert(const std::string & certPEM, std::string & outSubject,
+                                                                        std::string & outKeyId)
+{
+    // buffers and spans for storing crl signer delegator OR crl signer cert info
+    char subjectBuf[kMaxIssuerBase64Len]             = { 0 };
+    char skidBuf[2 * kAuthorityKeyIdentifierLength]  = { 0 };
+    uint8_t certDerBuf[kMax_x509_Certificate_Length] = { 0 };
+
+    MutableCharSpan subject(subjectBuf);
+    MutableCharSpan keyId(skidBuf);
+    MutableByteSpan certDER(certDerBuf);
+
+    ReturnLogErrorOnFailure(X509_PemToDer(certPEM, certDER));
+    ReturnErrorOnFailure(GetSubjectNameBase64Str(certDER, subject));
+    ReturnErrorOnFailure(GetSKIDHexStr(certDER, keyId));
+
+    outSubject = std::string(subject.data(), subject.size());
+    outKeyId   = std::string(keyId.data(), keyId.size());
+
+    return CHIP_NO_ERROR;
+}
+
+// Check if issuer and AKID matches with the crl signer OR crl signer delegator's subject and SKID
+bool TestDACRevocationDelegateImpl::CrossValidateCert(const Json::Value & revokedSet, const std::string & akidHexStr,
+                                                      const std::string & issuerNameBase64Str)
+{
+    std::string certPEM;
+    [[maybe_unused]] const char * certType;
```

Comment:
Why `[[maybe_unused]]`. Also, if you use std::string for some parts, why not use std::string for this as well?

Reply:
`certType` is only used in `ChipLogDetail`, and when the application is built with no detail logs, it is unused.